### PR TITLE
Enables git versioning for Windows/CMake builds

### PR DESCRIPTION
**Pull request by forkingpath**
Adding git revision module to CmakeMods & updating CMakeList to use it.

src/version.h will be generated each build, but it was previously 0.9something and replaced with 0.C as hardcoded in version.cpp, so I don't think anyone will mind.

### CODE REVIEW

File: src/version.cpp [hunk 1]
```diff
@@ -1,15 +1,5 @@
-#if (defined _WIN32 || defined WINDOWS || defined MINGW) && ! defined CROSS_LINUX && !defined _MSC_VER
-
-#ifndef VERSION
-#define VERSION "0.C" // FIXME: automatically generate VERSION based on git
```
Comment (kevingranade) [inline src/version.cpp]: This fallback is still needed for C::B and Visual Studio builds.

Comment (kevingranade) [inline src/version.cpp]: Possibly CMake and Make builds should inject a define to control importing version.h instead of us using build system matching.

**Comment by karlnp** [comment]
VERSION will still be defined, as I didn't delete version.h. It'll just be defined in  version.h instead of a weird override ifndef, which seems misplaced at the very least. 

Why don't we add a version.h with a static version and we can bump that or override it as necessary?

File: src/version.cpp (hunk 1 above)
Comment (karlnp) [inline src/version.cpp]: Why don't we add a version.h with a static version and we can bump that or override it as necessary?

Comment (karlnp) [inline src/version.cpp]: Sure.

**Comment by karlnp** [comment]
Is done.

**Comment by karlnp** [comment]
Jenkins is breaking with this one, but because it can't find ncurses?

**Pull request merged**